nc/source/component/Collider.cpp

```diff
@@ -7,29 +7,37 @@
 #include "ui/editor/Widgets.h"
 #endif
 
+#ifdef NC_EDITOR_ENABLED
+namespace
+{
+    bool IsUniformScale(const nc::Vector3& scale)
+    {
+        return nc::math::FloatEqual(scale.x, scale.y) && nc::math::FloatEqual(scale.y, scale.z);
+    }
+}
+#endif
+
 namespace nc
 {
     #ifdef NC_EDITOR_ENABLED
-    Collider::Collider(EntityHandle handle, ColliderType type, Vector3 offset, Vector3 scale)
+    Collider::Collider(EntityHandle handle, ColliderInfo info)
         : ComponentBase(handle),
+          m_type{info.type},
+          m_mask{info.mask},
           m_transformMatrix{GetComponent<Transform>(handle)->GetTransformationMatrix()},
-          m_boundingVolume{collider_detail::CreateBoundingVolume(type, offset, scale)},
-          m_type{type},
-          m_widgetModel{collider_detail::CreateWireframeModel(type)},
+          m_boundingVolume{collider_detail::CreateBoundingVolume(info.type, info.offset, info.scale)},
+          m_widgetModel{collider_detail::CreateWireframeModel(info.type)},
           m_selectedInEditor{false}
     {
-        IF_THROW(!HasNoZeroElement(scale), "Collider::Collider - Invalid scale(elements cannot be 0)");
-        IF_THROW(type == ColliderType::Sphere && scale - scale != Vector3::Zero(), "Collider::Collider - Sphere colliders do not support nonuniform scaling");
+        IF_THROW(!HasNoZeroElement(info.scale), "Collider::Collider - Invalid scale(elements cannot be 0)");
```

Comment:
Since this call is inverted so many times, it would be more efficient to create a HasAnyZeroElement that can jump out on the first FloatEqual(0.0f)